catalog/resource_connection.go

```diff
@@ -0,0 +1,100 @@
+package catalog
+
+import (
+	"context"
+
+	"github.com/databricks/databricks-sdk-go/service/catalog"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+// This structure contains the fields of catalog.UpdateConnection and catalog.CreateConnection
```

Comment:
We need better utilities for converting Go SDK structs to TF schemas so we don't need to inline stuff like this.